artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -1573,6 +1572,17 @@ public synchronized void cancelRedistributor() throws Exception {
 
    @Override
    public int getConsumerCount() {
+      // we don't want to count the redistributor, it is an internal transient entry in the consumer list
+      if (redistributor != null) {
```

Comment:
With this new code, we are adding still a perf penalty, as now with volatile reads we are going to main memory not cpu cache, this affects not just here, but where redistributor is used in many code places.

Reply:
@franz1981 can you review this?

Reply:
i guess it is needed, else it wouldnt be visible to other thread... problem is that's a perf change as all the reads will now be from main memory for it rather than just local cpu cache

Reply:
im shutting up, i dont think its that bad.....looks fine.

Reply:
yes im typing as im thinking apologies....

Reply:
no worries. your concerns and views are valid, there is a cost here. I will explore a little more if it is possible to encapsulate in queueConsumersImpl to avoid the volatile read.

Reply:
no its fine, you need it. i approved.